lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1989,21 +1990,7 @@ protected static function get_property_value( $styles, $path, $theme_json = null
 			return $value;
 		}
 
-		// Convert custom CSS properties.
-		$prefix     = 'var:';
-		$prefix_len = strlen( $prefix );
-		$token_in   = '|';
-		$token_out  = '--';
-		if ( 0 === strncmp( $value, $prefix, $prefix_len ) ) {
-			$unwrapped_name = str_replace(
-				$token_in,
-				$token_out,
-				substr( $value, $prefix_len )
-			);
-			$value          = "var(--wp--$unwrapped_name)";
-		}
-
-		return $value;
+		return self::convert_custom_properties( $value );
```

Comment:
If the constructor will take care of transforming data in the `var:preset|color|secondary` format to data in the `var(--wp--preset--color--secondary)` format, does any other code need to deal with that as well? Can we do instead?

```suggestion
		return $value;
```

Reply:
Well this is a logic I just extracted to a separate method, and `get_property_value` is used in `compute_style_properties`, `get_layout_styles` and `get_root_layout_rules`, so I'd rather no change the output of `get_property_value` in this PR.

Reply:
These are the "entry points" to this class:

- `__construct`: uses the `sanitize` method, which calls `sanitize_variables`.
- `remove_insecure_properties`: uses the `sanitize` method, which calls `sanitize_variables`.
- `get_from_editor_settings`: utility function, doesn't do anything with the internal `theme.json` dataset.
- `get_element_class_name`: utility function, doesn't do anything with the internal `theme.json` dataset.
- `scope_selector`: utility function, doesn't do anything with the internal `theme.json` dataset.

Any other method has the constructor or remove_insecure_properties as an ancestor. Given both sanitize the variables, I don't see how any other method can receive the internal format. Did I miss something? :thinking:

Reply:
ah, I got your idea incorrectly the first time, checking this, will apply the conversions to where it's needed.

Reply:
Just removed the [redundant usage](https://github.com/WordPress/gutenberg/pull/50366/commits/df276293029f46ee272e726181097a6a70a8f5bc) and everything seems to be fine, your analogy is correct since every method is using `$this->theme_json` which is already sanitized in the constructor.